packages/discovery-react-components/src/components/ResultsPagination/ResultsPagination.tsx

```diff
@@ -79,39 +140,13 @@ const ResultsPagination: FC<ResultsPaginationProps> = ({
     const pageFromOffset = Math.floor(actualOffset / actualPageSize) + 1;
     if (currentPage !== pageFromOffset) {
       setCurrentPage(pageFromOffset);
+      if (pageFromOffset === 1) {
+        reset({});
```

Comment:
Why call reset here? If I go to page 2 then hit Prev to go back to page 1, this will call reset?

Reply:
The `handleOnChange` function adjusts the `currentPage` and `offset` values at the same time, so `currentPage !== pageFromOffset` should never be true if the page change is being made via the pagination component itself. I'll double check that this portion of the code is not being hit in that case.

This piece is mostly to handle cases where something else is causing the offset to change outside of the user interacting with the pagination component (e.g. selecting a facet, creating a new query, etc.).